src/com/jgaap/util/DocumentHelper.java

```diff
@@ -71,6 +73,12 @@ static Document.Type getDocType(String filepath) {
 			return Document.Type.GENERIC;
 		}
 	}
+
+	static private String replaceCRLF(String text) {
+		// change CRLF sequences (\r\n, typical in Windows text files) to LF (\n, Linux & Mac)
+		text = text.replaceAll("\r(?=\n)", "");
```

Comment:
I think you meant for this line to be: `text = text.replaceAll("\r(?=\n)", "\n");`

Reply:
I think the empty string is correct because the regex only selects the \r in \r\n
![crlf1](https://user-images.githubusercontent.com/79772638/231923307-3d65c0f6-4d50-4111-8b89-538d71e5717f.png)
![crlf2](https://user-images.githubusercontent.com/79772638/231923315-0c9baf21-ea8c-4119-8c22-4d80391675bf.png)

Reply:
My concern is that there is a third option that is not being taken in to account here. As you state in the comment, Windows uses `\r\n` while Linux and Mac OS X (aka macOS) use `\n`. The third variant is `\r` which was used by Mac OS 9 and older. Even if this format is not as common today, it is possible that we could still encounter files that use it. The file does not even have to be sourced from a classic Mac OS system.

[Check out this text file.](https://github.com/evllabs/JGAAP/files/11228235/Old.Mac.Line.Ending.Sample.txt) I created it just now using Notepad++ on Windows. I haven't tested your regex yet to see if it works, but either way, this edge case is something that needs considered.

![image](https://user-images.githubusercontent.com/7663225/231927171-84dd065d-ad3b-4f86-bc26-e677fd0f88ed.png)

Reply:
I've tested your regex, and it doesn't seem to work for me. This one does, however:

```java
test = test.replaceAll("[\r\n]+", " ");
```

Sample Code for Testing
```java
public class RegexTest {
    public static void main(String args[]) {
      String test = "This is a\rtest to see\r\nhow\ndifferent types\r\nreact.";
      System.out.println(test);
      System.out.println();
      test = test.replaceAll("[\r\n]+", "<NEWLINE>");
      System.out.println(test);
    }
}
```

Sample Output
```
This is a
test to see
how
different types
react.

This is a<NEWLINE>test to see<NEWLINE>how<NEWLINE>different types<NEWLINE>react.
```

Reply:
I see. I didn't know \r was also used for line breaks lmao
How about this: ```(\r\n)|\r|\n```, preserving the number of line breaks and assuming no file mixes all types?
(Reason is this is applied to all files by default and normalizing multiple line breaks to single should be a canonicizer)

Tests
```
[CR only] "some carriage\rreturns\r\rto test\rstuff"
[LF only] "some linebreaks\n\nand some\nmore"
[CRLF]    "woah\r\nit's a\r\n\r\nparty\nover here"
[mixed]   "This is a\rtest to see\r\nhow\ndifferent types\r\nreact."
```
Output
```
some carriage<NEWLINE>returns<NEWLINE><NEWLINE>to test<NEWLINE>stuff
some linebreaks<NEWLINE><NEWLINE>and some<NEWLINE>more
woah<NEWLINE>it's a<NEWLINE><NEWLINE>party<NEWLINE>over here
This is a<NEWLINE>test to see<NEWLINE>how<NEWLINE>different types<NEWLINE>react.
```

Reply:
I didn't realize that the one I'd used normalizes line breaks. Given that, I agree that it would be more appropriate to use `(\r\n)|\r|\n`.